test/e2e/framework/test_context.go

```diff
@@ -36,7 +36,8 @@ import (
 )
 
 const (
-	defaultHost = "http://127.0.0.1:8080"
+	defaultHost        = "https://127.0.0.1:6443"
+	defaultBearerToken = "insecure-token"
```

Comment:
What happens if we randomize this, instead of using a const? Do the tests pass, or is there a problem with the way the process is forked for running the different services?

Reply:
I think using random token is OK, since the token is only generated once.

Reply:
hmm looks like we could not save random token in `framework.TestContext`, it would be difference across the tests.

Reply:
perhaps we could specify a random token via `--bearer-token` flag.

Reply:
ok. In that case, can we make the flag required, instead of defaulting to the insecure setup?

Reply:
If we made the flag required, we need to also update the job config https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml to pass this flag, otherwise the test won't pass.

I was considering could we use a default token first, then update the job config to pass `BUILD_ID`(https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables) as token to the job, and finally mark the token flag as required.

Reply:
> hmm looks like we could not save random token in framework.TestContext, it would be difference across the tests.

but this should not be a problem if we randomize the value of `framework.TestContext` if it equals `defaultBearerToken` once, at each run of the node suite [here](https://github.com/kubernetes/kubernetes/blob/3f585835d0a38462bb928454245ec01e834e53a0/test/e2e_node/e2e_node_suite_test.go#L120)?

Reply:
Actually I have tried this before, and it turns out that wouldn't help.

The root cause is that the test binary would invoke itself again with flag `--run-services-mode` to start dependant services, such as apiserver and etcd, like
```
I0926 21:26:13.050] I0926 21:24:05.791988     980 server.go:102] Starting server "services" with command "/tmp/node-e2e-20200926T212229/e2e_node.test --run-services-mode --test.timeout=24h0m0s --ginkgo.seed=1601155359 --ginkgo.focus=\\[NodeConformance\\] --ginkgo.skip=\\[Flaky\\]|\\[Slow\\]|\\[Serial\\] --ginkgo.parallel.node=1 --ginkgo.parallel.total=8 ... # remaining flags omitted
```
and the token that apiserver used is saved as file in this process.

Now that there are actually 2 process running, if we randomize the token inside the process, their value would be different.